### ISSUE TITLE
dream_rollout marks non-finite world-model predictions valid=True with weight 1.0, so NaN imagined transitions reach the control/GVF learners at full strength

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#454]
## Summary

`alberta_framework/core/dreaming.py`'s module docstring promises that every imagined step "must pass explicit warmup, model-error, and uncertainty gates before it reaches the control learner, so a poor or not-yet-trained model cannot corrupt control updates." The one-step guard (`GuardedDreamer.propose`) has an explicit `REJECT_NONFINITE` code and a finiteness conjunct in `accepted`. The rollout path (`dream_one_step` / `dream_rollout`) computes only

```python
valid = jnp.logical_and(rollout_state.active, jnp.logical_and(confidence_ok, error_ok))
```

with no finiteness check. `terminated = discount <= discount_floor` is `False` for NaN so the rollout stays active, and `next_observation = jnp.where(valid, world_prediction.next_observation, ...)` writes NaN into the scan carry, poisoning every later step too. `ImaginedTransition.valid` is the only mask the converters propagate (`imagined_transition_to_supervised_item`, `imagined_transition_to_gvf_item`, `imagined_rollout_to_gvf_items`, `imagined_rollout_to_sarsa_items`), so a NaN cumulant/reward/target ships with weight 1.0 and drives a full-strength control/GVF update to NaN — while the rollout's own diagnostics report a clean, fully valid horizon.

## Falsifier (deterministic, pristine `main` @ `b8a37de`; in-repo `ActionConditionedWorldModel` + `ActionConditionedDreamWorld` + `BehaviorModelDreamPolicy`, triggered by a non-finite anchor observation — the case `ActionConditionedWorldModel.predict` signals by returning an all-NaN prediction)

```
== dream_rollout from a non-finite anchor observation ==
valid          : [ True  True  True]
rewards        : [nan nan nan]
discounts      : [nan nan nan]
terminated     : [False False False]
gvf weights    : [1. 1. 1.]   gvf cumulants : [nan nan nan]
sarsa weights  : [1. 1. 1.]   sarsa rewards : [nan nan nan]

== same anchor through GuardedDreamer.propose (one-step guard, same module) ==
accepted: False  reject_code: 4 (REJECT_NONFINITE)
```

Expected: `dream_one_step` requires finite `next_observation`, `reward`, `discount`, `confidence`, and `model_error` for a step to be valid (mirroring the one-step guard), keeps the carried observation at its previous value, and stops the rollout — so every converter emits weight 0.0 for such steps.

## Dedup

`gh issue/pr list --search "dream" / "imagined" / "rollout" / "NaN transition accepted"`: adjacent only — #160/#161 (step-9 dream RNG, closed), PR #101 (unused inf channels in dream scores), PR #95 (world-model error EMA on inf transitions), #258 (NaN written on a *rejected* transition, the opposite direction). No open PR touches `core/dreaming.py`.

Fix PR follows (failing-test-first).
